### PR TITLE
imessage-exporter 1.1.0 (new formula)

### PR DESCRIPTION
**Pull request by unitof**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/imessage-exporter.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class ImessageExporter < Formula
+  desc "iMessage conversation database export and inspector"
+  homepage "https://github.com/ReagentX/imessage-exporter"
+  url "https://github.com/ReagentX/imessage-exporter/archive/refs/tags/1.1.0.tar.gz"
+  sha256 "b932862b5f4c1c607c4dd401f99d088b4e1291ce7d5c2a7d1c40da9c91478e97"
+  license "GPL-3"
```
Comment (SMillerDev) [inline Formula/imessage-exporter.rb:6]: Can it use later versions?

File: Formula/imessage-exporter.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  depends_on "rust" => :build
+
+  def install
+    # manifest set to 0.0.0 for some reason, matching upstream build behavior
+    # https://github.com/ReagentX/imessage-exporter/blob/develop/build.sh
+    inreplace "imessage-exporter/Cargo.toml", "version = \"0.0.0\"",
+                                              "version = \"#{version.to_s}\""
```
Comment (SMillerDev) [inline Formula/imessage-exporter.rb:14]: Can you ask upstream to properly name their releases?

Comment (unitof) [reply Formula/imessage-exporter.rb:14]: "Proper" would be to commit the most recent released version string, and bump with a tagged commit? (Right now, upstream's own build script injects the latest tagged release when it builds downloadable releases.)

File: Formula/imessage-exporter.rb (hunk 1 above)
Comment (unitof) [reply Formula/imessage-exporter.rb:6]: I don't believe so; don't see "or later" specified anywhere, unless I'm misunderstanding and that's the default? https://github.com/ReagentX/imessage-exporter/blob/develop/LICENSE

**Comment by unitof** [comment]
I can't figure out why my second test fails, happy to ask elsewhere if this is wrong forum.

I have confirmed but putting a `print` in the test and it definitely exits with `0` and outputs `Unable to launch: Database not found at /private/tmp/imessage-exporter-test-20230223-20634-2509a9/Library/Messages/chat.db`.

Yet every `brew test imessage-exporter` it seems to think it outputted an empty string:

```
Error: imessage-exporter: failed
An exception occurred within a child process:
  Minitest::Assertion: Expected /Unable\ to\ launch/ to match "".
```

**Comment by SMillerDev** [comment]
You can just tell the command to redirect it. `2>` IIRC

**Comment by chenrui333** [comment]
#126497

**Pull request closed without merging**